base/src/main/java/com/smartdevicelink/managers/screen/SoftButtonObject.java

```diff
@@ -73,17 +73,22 @@
      */
     public SoftButtonObject(@NonNull String name, @NonNull List<SoftButtonState> states, @NonNull String initialStateName, OnEventListener onEventListener) {
 
+        // If the list of states is empty, throw an error with DebugTool and return
+        if (states.isEmpty()) {
+            DebugTool.logError(TAG,"The state list is empty");
+            return;
+        }
         // Make sure there aren't two states with the same name
         if (hasTwoStatesOfSameName(states)) {
             DebugTool.logError(TAG, "Two states have the same name in states list for soft button object");
             return;
         }
 
         this.name = name;
-        this.states = states;
         this.currentStateName = initialStateName;
         this.buttonId = SOFT_BUTTON_ID_NOT_SET_VALUE;
         this.onEventListener = onEventListener;
+        this.states = states;
```

Comment:
Can we put this back to the same line it was previously on? this will help to reduce noise in the diff for changes that aren't really changes

Reply:
If you mean where it was when I was using setStates() in the constructor, I don't think I can without making the diff noisier, because now the check is happening there and the check obviously has to happen first to prevent assignment in the error case. This change is just reverting the states assignment back to how it was in develop before I forked. Maybe I could squash commits? But I think that might be more trouble than it's worth

Reply:
Just move `this.states = states;` back under `this.name= name;`

Reply:
Oh okay, will do